betamax/options.py

```diff
@@ -84,8 +86,9 @@ def items(self):
     def validate(self):
         for key, value in list(self.data.items()):
             if key not in Options.valid_options:
-                del self[key]
+                raise InvalidOption('{0} is not a valid option'.format(key))
             else:
                 is_valid = Options.valid_options[key]
                 if not is_valid(value):
-                    del self[key]
+                    raise validation_error_map[key]('{0} is not valid'
```

Comment:
Should we format in the value instead of the key here? Perhaps using `{0!r}` so it's `repr` formatted?